pkg/block/metrics.go

```diff
@@ -134,3 +137,59 @@ func (m *MetricsAdapter) GetRegion(ctx context.Context, storageID, storageNamesp
 func (m *MetricsAdapter) RuntimeStats() map[string]string {
 	return m.adapter.RuntimeStats()
 }
+
+type Histograms struct {
+	durationHistograms    *prometheus.HistogramVec
+	requestSizeHistograms *prometheus.HistogramVec
+}
+
+type AdapterMetricsHandler struct {
+	Histograms
+	metricsID *string
+}
+
+func InitHistograms(name string, withMetricsID bool) Histograms {
+	labelNames := []string{"operation", "error"}
+	if withMetricsID {
+		labelNames = append(labelNames, "metrics_id")
+	}
+	var durationHistograms = promauto.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name: name + "_operation_duration_seconds",
+			Help: "durations of outgoing " + name + " operations",
+		},
+		labelNames,
+	)
+	var requestSizeHistograms = promauto.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    name + "_operation_size_bytes",
+			Help:    "handled sizes of outgoing " + name + " operations",
+			Buckets: prometheus.ExponentialBuckets(1, 10, 10), //nolint: mnd
```

Comment:
TIL these the buckets we have. That's fairly nasty... (No action in this PR, of course.)